### PR TITLE
changes for including JSON instead of XML

### PR DESCRIPTION
**Pull request by hmunfru**
@jesuspg @jframos 
Still testing

### CODE REVIEW

**Comment by jframos** [comment]
LGTM

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 1]
```diff
@@ -593,22 +593,32 @@ public String obtainVMStatus(String vapp) throws VMStatusNotRetrievedException {
     public List<String> getIP(ClaudiaData claudiaData, String tierName, int replica, VM vm, String region)
             throws InfrastructureException {
         List<String> ips = new ArrayList<String>();
+        String response ="";
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: String response = "";

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 2]
```diff
@@ -593,22 +593,32 @@ public String obtainVMStatus(String vapp) throws VMStatusNotRetrievedException {
...
-                ips.add(ip);
+            response = openStackUtil.getServer(vm.getVmid(), region, token, vdc);
+
+            JSONObject jsonNodeServers = new JSONObject(response).getJSONObject("server");
+            JSONObject jsonAddress = jsonNodeServers.getJSONObject("addresses");
+            java.util.Iterator iter = jsonAddress.keys();
+            while(iter.hasNext()){
+                String key = (String)iter.next();
+                org.json.JSONArray value = jsonAddress.getJSONArray(key);
+                for (int i = 0; i < value.length(); i++) {  // **line 2**
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: what does it means? // **line 2**

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java [hunk 3]
```diff
@@ -593,22 +593,32 @@ public String obtainVMStatus(String vapp) throws VMStatusNotRetrievedException {
...
+                    ips.add(childJSONObject.getString("addr"));
+                }
             }
 
         } catch (OpenStackException oes) {
             String errorMessage = "Error interacting getting ip from OpenStack ";
             log.error(errorMessage);
             throw new InfrastructureException(errorMessage);
+        } catch (JSONException e) {
+            String errorMessage = "Error with Json ";
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/claudia/impl/ClaudiaClientOpenStackImpl.java]: Why do not you put e.getMessage() together with the string message that you obtain?

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/util/OpenStackUtilImpl.java [hunk 4]
```diff
@@ -697,17 +646,18 @@ public String getFloatingIP(PaasManagerUser user, String region) throws OpenStac
...
+    private String getFloatingIPFree(String response) throws JSONException {
         String floatingIP = null;
-        NodeList floatingIPs = findNodeList(xmlDoc, "floating_ip");
-        for (int i = 0; i < floatingIPs.getLength(); i++) {
-            Node floatingIPNode = floatingIPs.item(i);
-            if (findAttributeValueInNode(floatingIPNode, "instance_id").equals("None")) {
-                floatingIP = findAttributeValueInNode(floatingIPNode, "ip");
+        JSONArray jsonFloatingIps = new JSONObject(response).getJSONArray("floating_ips");
+
+        for (int i = 0; i < jsonFloatingIps.length(); i++) {  // **line 2**
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/util/OpenStackUtilImpl.java]: Could you be more explicit with the meaning of // **line 2**

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/util/OpenStackUtilImpl.java [hunk 5]
```diff
@@ -697,17 +646,18 @@ public String getFloatingIP(PaasManagerUser user, String region) throws OpenStac
...
      */
-    private String getFloatingIPFree(String xmlDoc) throws OpenStackException {
+    private String getFloatingIPFree(String response) throws JSONException {
         String floatingIP = null;
-        NodeList floatingIPs = findNodeList(xmlDoc, "floating_ip");
-        for (int i = 0; i < floatingIPs.getLength(); i++) {
-            Node floatingIPNode = floatingIPs.item(i);
-            if (findAttributeValueInNode(floatingIPNode, "instance_id").equals("None")) {
-                floatingIP = findAttributeValueInNode(floatingIPNode, "ip");
+        JSONArray jsonFloatingIps = new JSONObject(response).getJSONArray("floating_ips");
```
Comment (flopezag) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/util/OpenStackUtilImpl.java]: In general, why you do not put a try-catch around the operations getJSONArray in the different code blocks that I am seeing here?

**Comment by flopezag** [comment]
A general comment, I assume that it is necessary change the acceptance tests associate to this component too.

**Comment by flopezag** [comment]
Other impression, maybe I am wrong it that this changes are related only to obtain floating IPs in JSON format, but the rest of nova operations are taking into account? I assume that now all should be in JSON format, don't?

**Comment by hmunfru** [comment]
These changes are related to all request which were using XML. You can have a look at ClaudiaClientOpenStackImpl.java

File: core/src/main/java/com/telefonica/euro_iaas/paasmanager/util/OpenStackUtilImpl.java (hunk 5 above)
Comment (hmunfru) [inline core/src/main/java/com/telefonica/euro_iaas/paasmanager/util/OpenStackUtilImpl.java]: There is a throws JSONException in the method

**Comment by jframos** [comment]
Acceptance tests are not affected, because changes are related the way PaaSManager sends/receives requests to Nova. Features and messages of PaaSManager have not been modified.

**Comment by jesuspg** [comment]
LGTM

**Comment by flopezag** [comment]
LGTM

**Pull request merged**